### PR TITLE
Add Redis 2.0 package

### PR DESCRIPTION
**Pull request by BrennanConroy**
Copied code to new folders, will delete old packages in 3.0 branch after this is merged.

Hosted a redis instance and enabled the functional redis tests and verified they all pass. (Except for one that is asserting the wrong datetime format, but that one is broken with the old package too)

Fixes https://github.com/aspnet/Caching/issues/410

### CODE REVIEW

**Comment by Tratcher** [comment]
Did you fix the test with DateTime issue?

Rather than making this a complete copy, can you make it a move and have the old project file pull in the src files from the new location?

**Comment by BrennanConroy** [comment]
> Did you fix the test with DateTime issue?

No, I can take a quick glance at it.

> make it a move and have the old project file pull in the src files from the new location

Sure

File: build/dependencies.props [hunk 1]
```diff
@@ -19,6 +19,7 @@
     <MoqPackageVersion>4.7.49</MoqPackageVersion>
     <NETStandardLibrary20PackageVersion>2.0.3</NETStandardLibrary20PackageVersion>
     <StackExchangeRedisStrongNamePackageVersion>1.2.6</StackExchangeRedisStrongNamePackageVersion>
+    <StackExchangeRedisPackageVersion>2.0.513</StackExchangeRedisPackageVersion>
```
Comment (Tratcher) [inline build/dependencies.props]: @Eilon do you need to approve a new dependency version here?

**Comment by Tratcher** [comment]
Also copy over the baseline.netcore.json file.

File: build/dependencies.props (hunk 1 above)
Comment (Eilon) [reply build/dependencies.props]: This is approved.

File: src/Microsoft.Extensions.Caching.StackExchangeRedis/baseline.netcore.json [hunk 2]
```diff
@@ -0,0 +1,295 @@
+{
```
Comment (Eilon) [inline src/Microsoft.Extensions.Caching.StackExchangeRedis/baseline.netcore.json]: Brand new packages shouldn't have baselines, should they?

Comment (BrennanConroy) [reply src/Microsoft.Extensions.Caching.StackExchangeRedis/baseline.netcore.json]: "Brand new", it shouldn't have any API differences from the 2.2 package so we should be copying it over so api check verifies it.

Comment (Eilon) [reply src/Microsoft.Extensions.Caching.StackExchangeRedis/baseline.netcore.json]: Ah, nifty.

**Comment by BrennanConroy** [comment]
:up: :date: Changed namespaces and changed DI call to `AddDistributedStackExchangeRedisCache`

File: src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.StackExchangeRedis;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up Redis distributed cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class RedisCacheServiceCollectionExtensions
```
Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs]: This class name needs to change

File: src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        /// <summary>
+        /// Adds Redis distributed caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="setupAction">An <see cref="Action{RedisCacheOptions}"/> to configure the provided
+        /// <see cref="RedisCacheOptions"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddDistributedStackExchangeRedisCache(this IServiceCollection services, Action<RedisCacheOptions> setupAction)
```
Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs]: AddStackExchangeRedisCache?

Comment (BrennanConroy) [reply src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs]: Who would make the decision here?

Comment (Tratcher) [reply src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs]: @Eilon or @DamianEdwards

Comment (Eilon) [reply src/Microsoft.Extensions.Caching.StackExchangeRedis/RedisCacheServiceCollectionExtensions.cs]: Per discussion, we can drop the `Distributed` from here, though I'm fine as-is, or even `AddStackExchangeRedisDistributedCache`

**Comment by moodmosaic** [comment]
Is there a way to get this from a nightly build or preview?

**Comment by Tratcher** [comment]
The nightly's are cumbersome but [preview3](https://github.com/aspnet/Announcements/issues/307) should be available soon.

**Comment by Tratcher** [comment]
Yes. https://github.com/aspnet/SignalR/pull/3089

**Comment by Tratcher** [comment]
FYI the nighty instructions are [here](https://github.com/aspnet/aspnetcore#daily-builds) if you're brave.

**Pull request merged**